### PR TITLE
Adding folly

### PR DESCRIPTION
**Pull request by pearu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pearu** [comment]
@conda-forge/help-c-cpp, please review.

File: recipes/folly/build.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Build shared library
+export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DBUILD_SHARED_LIBS=ON"
+
+# Resolves #error No clock_gettime(3) compatibility wrapper available for this platform.
+export CXXFLAGS="$CXXFLAGS -DFOLLY_HAVE_CLOCK_GETTIME=1"
+
+mkdir -p _build
+cd _build
+
+cmake -Wno-dev \
```
Comment (xhochy) [inline recipes/folly/build.sh:28]:
```suggestion
cmake ${CMAKE_ARGS} -Wno-dev \
```

File: recipes/folly/build.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  -DCMAKE_PREFIX_PATH=$PREFIX \
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (xhochy) [inline recipes/folly/build.sh:30]: Already part of `CMAKE_ARGS`
```suggestion
```

File: recipes/folly/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  run:
+    - boost-cpp
+    - double-conversion
+    - fmt
+    - gflags
+    - glog
+    - libaio
+    - libevent
+    - libsodium
+    - snappy
```
Comment (xhochy) [inline recipes/folly/meta.yaml:49]: They should all come in via their `run_exports`

File: recipes/folly/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  run_constrained:
+    - boost-cpp
```
Comment (xhochy) [inline recipes/folly/meta.yaml:51]: This is already part of run
```suggestion
```

File: recipes/folly/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+test:
+  requires:
+    - {{ compiler('cxx') }}
+    - jemalloc
+  commands:
+    - test -f ${PREFIX}/lib/libfolly.so
+    - test -f ${PREFIX}/include/folly/folly-config.h
+    - $CXX -I$PREFIX/include -L$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -lfolly $RECIPE_DIR/arena_test.c -o arena_test
```
Comment (xhochy) [inline recipes/folly/meta.yaml:60]:
```suggestion
    - $CXX $CXXFLAGS -lfolly $RECIPE_DIR/arena_test.c -o arena_test
```

File: recipes/folly/build.sh (hunk 1 above)
Comment (pearu) [reply recipes/folly/build.sh]: Applied.

File: recipes/folly/build.sh (hunk 2 above)
Comment (pearu) [reply recipes/folly/build.sh]: Applied.

File: recipes/folly/meta.yaml (hunk 3 above)
Comment (pearu) [reply recipes/folly/meta.yaml]: Fixed: removed the listed dependencies from `run`

File: recipes/folly/meta.yaml (hunk 4 above)
Comment (pearu) [reply recipes/folly/meta.yaml]: Applied.

File: recipes/folly/meta.yaml (hunk 5 above)
Comment (pearu) [reply recipes/folly/meta.yaml]: Applied.

**Comment by pearu** [comment]
@xhochy thanks for the review, I have applied your suggestions and cleaned up the build.sh.

File: recipes/folly/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+package:
+  name: folly
+  version: {{ version }}
+
+source:
+  url: https://github.com/facebook/folly/archive/v{{ version }}.tar.gz
+  sha256: 2f9b0ff1af934ca715028462985b6c22004f2599af157bab475857ffc9a5688d
+
+build:
+  number: {{ number }}
```
[the remark this answers is not in the document]
Comment (pearu) [reply recipes/folly/meta.yaml]: Done.

**Pull request merged**